src/main.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/parser"
+	"go/token"
+	"log"
+	"os"
+)
+
+type logger interface {
+	log(format string, args ...interface{}) error
+}
+
+type checker interface {
+	run(fset *token.FileSet, logger logger) error
```

Comment:
We can introduce `lint.Context` type that will hold both `fset`, and `logger`, and perhaps even some (or all) command line flags that we will almost certainly need inside some checkers.
(This means that we need `src/lint/lint.go` file that defines `Context` type; you may call that `Config`, if you find it's more fitting.)